src/main/java/edu/harvard/iq/dataverse/api/MakeDataCountApi.java

```diff
@@ -170,7 +170,13 @@ public Response updateCitationsForDataset(@PathParam("id") String id) throws Mal
             } while (nextPage == true);
             JsonArray allData = dataBuilder.build();
             List<DatasetExternalCitations> datasetExternalCitations = datasetExternalCitationsService.parseCitations(allData);
-
+            /*
+             * ToDo: If this is the only source of citations, we should remove all the existing ones for the dataset and repopuate them.
```

Comment:
I'm confused by this comment. When I look into at `datasetExternalCitationsService.save(dm)` I see logic in there about repopulating and a comment that says "Replace existing if necessary". Some variables in that method were renamed in this pull request so it seems to have been examined.

That said there's more below (hasPart, etc.) that I'm not entirely following. And this is just a comment, not a code change. 😄 

If there's a bug, it might be better to not add this comment and simply create an issue to handle it separately.

Reply:
The save method only replaces an existing entry if a new one that matches is found. If a citation ever goes away, it will never get deleted. While one could argue that citations won't go away, the practical example from QDR is that from ~ v.4.17 we sent hasPart info to DataCite and that was originally pulled back as a citation. PRs now in v4.20 now filter what comes back from DataCite to remove these and other 'non-citation' relationships, but if the upload citations method was run and db entries created, they won't get deleted, and future changes to the list of relationships to consider also won't clear old db entries.